src/model/account/PublicAccount.ts

```diff
@@ -53,6 +54,48 @@ export class PublicAccount {
         return new PublicAccount(publicKey, address);
     }
 
+    /**
+     * Verify a signature.
+     *
+     * @param {string} publicKey - The public key to use for verification.
+     * @param {string} data - The data to verify.
+     * @param {string} signature - The signature to verify.
+     *
+     * @return {boolean}  - True if the signature is valid, false otherwise.
+     */
+    static verifySignature(publicKey: string, data: string, signature: string): boolean {
```

Comment:
changing `publicKey: string` by `publicAccount: PublicAccount` will remove the next code,

```javascript
if (publicKey.length !== 64 && publicKey.length !== 66) {
    throw new Error('Not a valid public key');
}
```
making the design safer.

Reply:
`signature` has some constrains too, I'm thinking in create a class to represent the signature and add there the invariants. It should remove the:

```javascript
if (signature.length !== 128) {
    throw new Error('Signature length is incorrect !');
}
// ...
const _signature = convert.hexToUint8(signature);
```
Also, it hasn't been checked that the signature is an hexadecimal string.

Reply:
> Also, it hasn't been checked that the signature is an hexadecimal string

well, this "if condition" below made sure that the signature must be hexadecimal

```
if (convert.isHexString(signature)) {
        throw new Error('Signature must be hexadecimal only');
}
```